### PR TITLE
Validator update after moving the amp-story-consent JSON configuration.

### PR DESCRIPTION
**Pull request by gmajoulet**
Validator update after moving the amp-story-consent JSON configuration.

Context: https://github.com/ampproject/amphtml/pull/15366/files/f48b29ca4203dd4cfce001b9555128d307551c7b#r189401119

Linked to #15473

### CODE REVIEW

File: extensions/amp-story/validator-amp-story.protoascii [hunk 1]
```diff
@@ -323,18 +323,41 @@ tags: {  # <amp-story-consent>
   html_format: AMP
   tag_name: "AMP-STORY-CONSENT"
   requires_extension: "amp-consent"
+  requires: "amp-story-consent extension .json script"
```
Comment (zhouyx) [inline extensions/amp-story/validator-amp-story.protoascii]: I thought we don't need this? Did you say that amp-story-0.1.js include the amp-story-consent logic?

File: extensions/amp-story/validator-amp-story.protoascii [hunk 2]
```diff
@@ -319,17 +321,43 @@ tags: {  # <amp-story-bookend>
     }
   }
 }
+tags: {  # amp-story-consent (json)
```
Comment (zhouyx) [inline extensions/amp-story/validator-amp-story.protoascii]: I am not familiar with the validator rules : ) would https://github.com/ampproject/amphtml/pull/15474/files#diff-2f32ef70db93f072c60fc1a70694c7bbR359 does the job?

Comment (gmajoulet) [reply extensions/amp-story/validator-amp-story.protoascii]: This rule ensures it's a ``<script>`` with the ``application/json`` attribute, has ``amp-story-consent`` as a parent, etc : )

File: extensions/amp-story/validator-amp-story.protoascii [hunk 3]
```diff
@@ -175,6 +175,7 @@ tags: {  # <amp-story-page>
 tags: {  # <amp-story-grid-layer>
   html_format: AMP
   tag_name: "AMP-STORY-GRID-LAYER"
+  requires_extension: "amp-story"
```
Comment (zhouyx) [inline extensions/amp-story/validator-amp-story.protoascii]: unrelated changes?

File: extensions/amp-story/validator-amp-story.protoascii [hunk 4]
```diff
@@ -319,17 +321,43 @@ tags: {  # <amp-story-bookend>
...
+    }
+  }
+}
 tags: {  # <amp-story-consent>
   html_format: AMP
   tag_name: "AMP-STORY-CONSENT"
   requires_extension: "amp-consent"
+  requires_extension: "amp-story"
   mandatory_parent: "AMP-CONSENT"
+  requires: "amp-story-consent extension .json script"
```
Comment (zhouyx) [inline extensions/amp-story/validator-amp-story.protoascii]: what does this do?

File: extensions/amp-story/validator-amp-story.protoascii (hunk 3 above)
Comment (honeybadgerdontcare) [reply extensions/amp-story/validator-amp-story.protoascii]: these are enforced via the mandatory_parent (because the parent has it) but these should have it too.

File: extensions/amp-story/validator-amp-story.protoascii (hunk 4 above)
Comment (honeybadgerdontcare) [reply extensions/amp-story/validator-amp-story.protoascii]: this requires the json script to be present within amp-story-consent.

File: extensions/amp-story/validator-amp-story.protoascii (hunk 3 above)
Comment (honeybadgerdontcare) [reply extensions/amp-story/validator-amp-story.protoascii]: i'll remove and put in a separate pr

File: extensions/amp-story/validator-amp-story.protoascii (hunk 4 above)
Comment (zhouyx) [reply extensions/amp-story/validator-amp-story.protoascii]: I see. Thanks!
